app/scripts/controllers/app-state.js

```diff
@@ -191,4 +194,27 @@ export default class AppStateController extends EventEmitter {
   setBrowserEnvironment(os, browser) {
     this.store.updateState({ browserEnvironment: { os, browser } });
   }
+
+  /**
+   * Adds a pollingToken for a given environmentType
+   * @returns {void}
+   */
+  addPollingToken(pollingToken, pollingTokenType) {
+    this.store.updateState({
+      [pollingTokenType]: [
+        ...this.store.getState()[pollingTokenType],
+        pollingToken,
+      ],
+    });
+  }
+
+  removePollingToken(pollingToken, pollingTokenType) {
+    this.store.updateState({
+      [pollingTokenType]: [
```

Comment:
Nit: Could be a little easier to read if we put `this.store.getState()[pollingTokenType]` in a variable at the start of the function, and then reference the variable here